test/test_autograd.py

```diff
@@ -3333,6 +3333,343 @@ def test_checkpointing(self):
+                with torch.set_grad_enabled(grad_mode):
+                    return inp.unbind()
+
+            def maybe_check_raise(fn, should_raise):
+                self.assertTrue(should_raise is None or isinstance(should_raise, str))
+                if should_raise is not None:
+                    with self.assertRaisesRegex(RuntimeError, should_raise):
+                        fn()
+                else:
+                    fn()
+
+            # Are they differentiable views?
+            out = get_view_as()
+            self.assertTrue(out._is_view() == is_view)
+
+            out = get_unbind()
+            self.assertTrue(out[0]._is_view() == is_view)
+            self.assertTrue(out[1]._is_view() == is_view)
+
+            # Are inplace allowed?
+            out = get_view_as()
+            maybe_check_raise(lambda: out.add_(1), should_raise_tuple[0])
+
+            out = get_unbind()
+            maybe_check_raise(lambda: out[0].add_(1), should_raise_tuple[1])
+            maybe_check_raise(lambda: out[1].add_(1), should_raise_tuple[2])
+
+
+        # should_raise contains None if it should not raise
+        # should_raise contains a string of the error if it should raise
+        # The 3 elements are for view_as, first output of unbind and second output of unbind
+        run_test(grad_mode=True, requires_grad=False, is_view=True,
+                 should_raise_tuple=(None, None, None))
+        # TODO: Second should_raise should not be None below
+        run_test(grad_mode=True, requires_grad=True, is_view=True,
+                 should_raise_tuple=(None, None, "diff_view_meta->output_nr_ == 0 INTERNAL ASSERT FAILED"))
+        # TODO: views require gradients when created in no_grad mode but their grad_fn is not populated
+        leaf_grad_err = "a leaf Variable that requires grad has been used in an in-place operation."
+        run_test(grad_mode=False, requires_grad=True, is_view=True,
+                 should_raise_tuple=(leaf_grad_err, leaf_grad_err, leaf_grad_err))
+        run_test(grad_mode=False, requires_grad=False, is_view=True,
+                 should_raise_tuple=(None, None, None))
+
+
+    def test_autograd_views_python(self):
+        # This is not necessarily the absolute correct behavior, but this is the current
+        # one. This test is here to make sure that any change to this behavior is detected
+        # and not silent. The TODOs below mark the places with unexpected behavior.
+        # Note that any change in these test will be BC-breaking and should be done carefully.
+
+        # I) This checks the autograd.Function behavior when we return one or multiple outputs
+        # while one of these is an input, a view of an input or of a temporary tensor.
+        # II) This checks that multiples views in the forward are properly traced and how they
+        # behave with respect to inplace operations.
+
+
+        # I) Simple views
+        bw_called = [0]
+        ga_nz = [False]
```

Comment:
I think it would be very helpful for readers if it was explained what these indicator variables represent